### PR TITLE
config(server): Add PG pool max connection config.

### PR DESCRIPTION
**Pull request by charlesbthomas**

### CODE REVIEW

File: packages/server/src/config.ts [hunk 1]
```diff
@@ -112,6 +112,8 @@ export interface MedplumDatabaseConfig {
   ssl?: MedplumDatabaseSslConfig;
   queryTimeout?: number;
   runMigrations?: boolean;
+  maxThreads?: number;
```
Comment (codyebberson) [inline packages/server/src/config.ts:115]: How would you feel about `maxClients` or `maxConnections`?  

Sorry to be picky about this, but config setting names live forever, and naming things is hard 🙃

File: packages/server/src/database.ts [hunk 2]
```diff
@@ -47,7 +47,8 @@ async function initPool(config: MedplumDatabaseConfig, proxyEndpoint: string | u
     user: config.username,
     password: config.password,
     ssl: config.ssl,
-    max: 100,
+    max: config.maxThreads ?? 100,
+    min: config.minThreads,
```
Comment (codyebberson) [inline packages/server/src/database.ts:51]: Hm, this is interesting.

`min` is not in the docs: https://node-postgres.com/apis/pool

`min` is not in the pool source: https://github.com/brianc/node-postgres/blob/master/packages/pg-pool/index.js

But `min` is in the types: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pg/index.d.ts#L46

I don't have any objections to `min`, I just want to make sure we understand what it does.

File: packages/server/src/config.ts (hunk 1 above)
Comment (charlesbthomas) [reply packages/server/src/config.ts:115]: I am good with maxConnections!

File: packages/server/src/database.ts (hunk 2 above)
Comment (charlesbthomas) [reply packages/server/src/database.ts:51]: Wow, that's kind of shocking, min connections is a pretty common concept for connection thread pools IME. But I am totally good to exclude it from this PR; generally speaking max connections is more critical.

File: packages/server/src/config.ts (hunk 1 above)
Comment (charlesbthomas) [reply packages/server/src/config.ts]: Done!

File: packages/server/src/database.ts (hunk 2 above)
Comment (charlesbthomas) [reply packages/server/src/database.ts]: Removed!

**Review by codyebberson — APPROVED** [review]

File: packages/server/src/database.ts (hunk 2 above)
Comment (codyebberson) [reply packages/server/src/database.ts]: Agree that it is a common feature of connection pools.

The `pg` library has been good to us, but it is showing signs of neglect and decay.

We have run some internal experiments with [`postgres.js`](https://github.com/porsager/postgres), which is modernized, has better Postgres pipelining support, and is generally much faster.

image

(lower is better)

Source: https://github.com/porsager/postgres-benchmarks

It is not as widely used though, and is not as actively maintained as we would like.
image

https://npmtrends.com/pg-vs-postgres

**Pull request merged**